pyartcd/pyartcd/pipelines/ocp4_konflux.py

```diff
@@ -625,20 +625,22 @@ async def mirror_images(self):
         # Get the list of successful builds
         builds_to_mirror = [entry for entry in record_log['image_build_konflux'] if not int(entry['status'])]
 
-        for build in builds_to_mirror:
+        async def sync_build(build):
             release = build["nvrs"].split("-")[-1]
             image_pullspec = build["image_pullspec"]
 
             if self.assembly != "stream":
                 LOGGER.info(f"Not syncing {image_pullspec} because assembly {self.assembly} != stream")
-                continue
+                return
 
             if is_release_embargoed(release=release, build_system="konflux"):
                 LOGGER.info(f"Not syncing {image_pullspec} because it is in an embargoed release")
-                continue
+                return
 
             await sync_to_quay(image_pullspec, KONFLUX_ART_IMAGES_SHARE)
 
+        await asyncio.gather(*[sync_build(build) for build in builds_to_mirror])
```

Comment:
I would add a `@limit_concurrency` decorator around `sync_to_quay()` as we normally do to limit parallel `oc` calls

Reply:
Thank you @locriandev - updated